### PR TITLE
wip rewrite in typescript

### PR DESCRIPTION
**Pull request by jumoog**
## Summary by Sourcery

Rewrite the IntroSkipper plugin in TypeScript to improve code maintainability and type safety. Update CI workflows to support TypeScript compilation.

New Features:
- Rewrite the IntroSkipper plugin in TypeScript, introducing type safety and improved code structure.

Enhancements:
- Add TypeScript configuration file (tsconfig.json) to the project, enabling TypeScript compilation and type checking.

CI:
- Update CI workflows to include TypeScript setup and compilation steps, ensuring TypeScript code is compiled during the build process.

### CODE REVIEW

File: ConfusedPolarBear.Plugin.IntroSkipper/Configuration/inject.ts [hunk 1]
```diff
@@ -1,57 +1,111 @@
...
+}
+
+declare const ApiClient: ApiClient;
+
+interface SkipperData {
+    Introduction?: Segement
+    Credits?: Segement
+}
+
+interface Segement {
```
Comment (Gauvino) [inline ConfusedPolarBear.Plugin.IntroSkipper/Configuration/inject.ts:13]: Is this normal ?
```suggestion
interface Segment {
```
It should be like this right ?

**Review by Gauvino — COMMENTED** [review]
Same on other part of the code

**Comment by AbandonedCart** [comment]
Bob the Builder: The Pull Request

**Comment by jumoog** [comment]
@sourcery-ai review

**Comment by jumoog** [comment]
@sourcery-ai bro gimme the code

**Pull request closed without merging**